Build failure: libcxxabi on Darwin

### Steps To Reproduce

Steps to reproduce the behavior:
1. build libcxxabi on Darwin

### Build log

```
building '/nix/store/ac8awd5c8fj8n0ya3bh5adkcz6mljijz-libcxxabi-15.0.7.drv'...
unpacking sources
unpacking source archive /nix/store/fgdzkqv6f72axj9zgkv94lm9f9wqmns3-libcxxabi-src-15.0.7
source root is libcxxabi-src-15.0.7/runtimes
patching sources
applying patch /nix/store/4hxmswl71hy1dc037qxj2nb6g3hph25b-gnu-install-dirs.patch
patching file CMakeLists.txt
applying patch /nix/store/wjqqjq7dcfr4yslnd21zwsm60rs822xz-e6a0800532bb409f6d1c62f3698bdd6994a877dc.patch
patching file CMakeLists.txt
patching file test/CMakeLists.txt
configuring
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/share/doc/Runtimes -DCMAKE_INSTALL_INFODIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/xvsi8zixbqcvh4ypw69dzy9w5a2ikldc-libcxxabi-15.0.7-dev/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/xvsi8zixbqcvh4ypw69dzy9w5a2ikldc-libcxxabi-15.0.7-dev/include -DCMAKE_INSTALL_SBINDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/rwb88nwzbrsk2k0ldf1b5scd5yyrfgl9-cctools-binutils-darwin-11.1.0-973.0.1/bin/strip -DCMAKE_RANLIB=/nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/ranlib -DCMAKE_AR=/nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7 -DLLVM_ENABLE_RUNTIMES=libcxxabi -DLIBCXXABI_LIBCXX_INCLUDES=/nix/store/kd08z58zhhpvrww8a2f74dx1dr3frggb-libcxx-headers-15.0.7/include/c++/v1 -DCMAKE_CXX_COMPILER_WORKS=ON -DCMAKE_EXE_LINKER_FLAGS=-nostdlib -DCMAKE_SHARED_LINKER_FLAGS=-nostdlib
-- The C compiler identification is Clang 15.0.7
-- The CXX compiler identification is Clang 15.0.7
-- The ASM compiler identification is Clang with GNU-like command-line
-- Found assembler: /nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang
-- Check for working C compiler: /nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang - broken
CMake Error at /nix/store/inh6dp2sykgdnh6g4sblmng1s89vz08c-cmake-3.26.4/share/cmake-3.26/Modules/CMakeTestCCompiler.cmake:67 (message):
  The C compiler

    "/nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /tmp/nix-build-libcxxabi-15.0.7.drv-0/libcxxabi-src-15.0.7/runtimes/build/CMakeFiles/CMakeScratch/TryCompile-t6lOIn

    Run Build Command(s):/nix/store/mqyp4gwyppmcykj3vk3ig9z1lgnvrrji-ninja-1.11.1/bin/ninja -v cmTC_72306 && [1/2] /nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang    -MD -MT CMakeFiles/cmTC_72306.dir/testCCompiler.c.o -MF CMakeFiles/cmTC_72306.dir/testCCompiler.c.o.d -o CMakeFiles/cmTC_72306.dir/testCCompiler.c.o -c /tmp/nix-build-libcxxabi-15.0.7.drv-0/libcxxabi-src-15.0.7/runtimes/build/CMakeFiles/CMakeScratch/TryCompile-t6lOIn/testCCompiler.c
    [2/2] : && /nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang  -Wl,-search_paths_first -Wl,-headerpad_max_install_names -nostdlib CMakeFiles/cmTC_72306.dir/testCCompiler.c.o -o cmTC_72306   && :
    FAILED: cmTC_72306
    : && /nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang  -Wl,-search_paths_first -Wl,-headerpad_max_install_names -nostdlib CMakeFiles/cmTC_72306.dir/testCCompiler.c.o -o cmTC_72306   && :
    ld: dynamic main executables must link with libSystem.dylib for architecture x86_64
    clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
    ninja: build stopped: subcommand failed.


  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:3 (project)


-- Configuring incomplete, errors occurred!
error: builder for '/nix/store/ac8awd5c8fj8n0ya3bh5adkcz6mljijz-libcxxabi-15.0.7.drv' failed with exit code 1
```

### Additional context

Could this be caused by https://github.com/NixOS/nixpkgs/pull/246577 ? (@pwaller)

### Notify maintainers
@vlstill 

### Metadata

```
- system: `"x86_64-darwin"`
 - host os: `Darwin 22.5.0, macOS 10.16`
 - multi-user?: `no`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.10.3`
 - channels(djm): `"home-manager, nixpkgs, unstable"`
 - nixpkgs: `/Users/djm/.nix-defexpr/channels/nixpkgs`
 ```

It originally happened when doing `home-manager switch`, as a dependency of mpv, so this reproduces it:
```
~/src/nixpkgs $ nix-build -A mpv
these 41 derivations will be built:
  /nix/store/8di4w8xv41si6wwncd55p7w1np9fcbca-xcrun.drv
  /nix/store/ac8awd5c8fj8n0ya3bh5adkcz6mljijz-libcxxabi-15.0.7.drv
  /nix/store/3x3qxzgdbdzk706651wyhb3441sscr9p-compiler-rt-libc-15.0.7.drv
  /nix/store/n1ssw0pqiwb41gmi41zplqsfvl0zhcz0-clang-wrapper-15.0.7.drv
  /nix/store/59drfx8xm7c2lni83ipj4swiw6blwyaf-stdenv-darwin.drv
  /nix/store/l39vwfh62bj4v9bz2wbjdc8kynzhbv2x-lua-5.2.4-env.drv
  /nix/store/ns5s6v4hxxzzn8zi26zgjaj0slgar23i-xcrun.drv
  /nix/store/60mg7415kjw8mw8fxsjikm4798hbmapp-nix-swift-make-clang-wrapper.drv
  /nix/store/45zmljfca6nydxdwiririx1gjigrxj5k-nix.xcconfig.drv
  /nix/store/9b6c8ahml4ka2czck2gzcrqbimg6cfmh-Toolchains.drv
  /nix/store/hz2hv93n8bfg5v3m1h6wd40c2qlnas1f-stdenv-darwin.drv
  /nix/store/hgiw06yz04jbjxzfm2x6h7vvbs2bfmfv-xcbuild-0.1.2-pre.drv
  /nix/store/mz41n21lh7y8vm9y1iang2vwgcjji2kl-xcode-select.drv
  /nix/store/r7wiyi7blxn1hmiz9qb44w9fclymj4a1-xcrun.drv
  /nix/store/75vyd1li5va600bx3k99xw0msmma5sy0-xcodebuild-0.1.2-pre.drv
  /nix/store/g5rxzpdal845rh689mnsahflp7dr2rih-swift-darwin-plistbuddy-workaround.patch.drv
  /nix/store/ibn6vnib20jms9hwam74w88jb89awmlj-libtool.drv
  /nix/store/lqcq8hzfh5s1adc5jfbwdjqhp041k8d6-swift-prevent-sdk-dirs-warning.patch.drv
  /nix/store/smw0srshldxg2cwy28ch6jy93ksbwz10-apple-swift-core.drv
  /nix/store/w315q4a0j15l31mq75csf3bfnwwrk3zy-swift-wrapper.sh.drv
  /nix/store/wf4kwk1gq8jhz08nwhgz11m3z1b07k7z-nix-swift-make-swift-wrapper.drv
  /nix/store/q6acbkakd3ly1migpi614x514m7rmknk-swift-5.8.drv
  /nix/store/r4v4nbynjq5qcz4caa745jqc6kkc6z65-swift-wrapper-5.8.drv
  /nix/store/1qhgcg44j7a1vkxk0a624p9rcsn0wpja-swift-corelibs-xctest-5.8.drv
  /nix/store/4jc6gigp294pqq7vmzgdc7xq8l924bq4-prevent-sdk-dirs-warnings.patch.drv
  /nix/store/355qhzrysxwbisdwbmkijwqwlxc0zfnn-swiftpm-cctools.drv
  /nix/store/28hyq1l9w3331wmd1nha1z75zax2innb-swift-argument-parser.drv
  /nix/store/dazgj6vj1yp2w2byy0qkq3si14dvx4g3-llbuild.drv
  /nix/store/6wqbs1lwgsssh4d9la1qvla92iq6ydzq-Yams.drv
  /nix/store/j5aj2q7lg8wm3zh9ymw2zlr9wpk0f01l-swift-system.drv
  /nix/store/l4l7dm9hr7wm5n33bnwz4gsgp9479lpc-swift-tools-support-core.drv
  /nix/store/gdyig3j7lmj3ismbv3bnxi8qq28dpiyn-swift-driver.drv
  /nix/store/mcvz6piv5am6f8xm8vm8bx5gzgx3fhpm-swift-crypto.drv
  /nix/store/svbwgpkdvhswwkpqv3yzxgv9s4ll1c37-fix-stdlib-path.patch.drv
  /nix/store/ywmar5kmklrvva00zh7zwklgvhwy0y1i-swift-collections.drv
  /nix/store/i72lazrmma1ln2vyvmdnkg5idw9id9rs-swiftpm-bootstrap-5.8.drv
  /nix/store/ivsczq1hbv3q5w4rpmbfaiax2np7ijnr-swiftpm-5.8.drv
  /nix/store/fj72c2jch1xcri1nc8zkqfd5adjq4y7j-swift-driver-5.8.drv
  /nix/store/slgq6qia7pfkfb4m316wh2hgd34z5q7m-swift-wrapper-5.8.drv
  /nix/store/7dkwf3rkhlcjvm8pap8f12y99g86bjj4-mpv-0.36.0.drv
  /nix/store/0gdqb9idj7abnz5b3j6v9qpaj37l1cxy-mpv-with-scripts-0.36.0.drv
building '/nix/store/l39vwfh62bj4v9bz2wbjdc8kynzhbv2x-lua-5.2.4-env.drv'...
created 11 symlinks in user environment
building '/nix/store/ac8awd5c8fj8n0ya3bh5adkcz6mljijz-libcxxabi-15.0.7.drv'...
unpacking sources
unpacking source archive /nix/store/fgdzkqv6f72axj9zgkv94lm9f9wqmns3-libcxxabi-src-15.0.7
source root is libcxxabi-src-15.0.7/runtimes
patching sources
applying patch /nix/store/4hxmswl71hy1dc037qxj2nb6g3hph25b-gnu-install-dirs.patch
patching file CMakeLists.txt
applying patch /nix/store/wjqqjq7dcfr4yslnd21zwsm60rs822xz-e6a0800532bb409f6d1c62f3698bdd6994a877dc.patch
patching file CMakeLists.txt
patching file test/CMakeLists.txt
configuring
fixing cmake files...
cmake flags: -GNinja -DCMAKE_FIND_USE_SYSTEM_PACKAGE_REGISTRY=OFF -DCMAKE_FIND_USE_PACKAGE_REGISTRY=OFF -DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTING=OFF -DCMAKE_INSTALL_LOCALEDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/share/locale -DCMAKE_INSTALL_LIBEXECDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/libexec -DCMAKE_INSTALL_LIBDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/lib -DCMAKE_INSTALL_DOCDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/share/doc/Runtimes -DCMAKE_INSTALL_INFODIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/share/info -DCMAKE_INSTALL_MANDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/share/man -DCMAKE_INSTALL_OLDINCLUDEDIR=/nix/store/xvsi8zixbqcvh4ypw69dzy9w5a2ikldc-libcxxabi-15.0.7-dev/include -DCMAKE_INSTALL_INCLUDEDIR=/nix/store/xvsi8zixbqcvh4ypw69dzy9w5a2ikldc-libcxxabi-15.0.7-dev/include -DCMAKE_INSTALL_SBINDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/sbin -DCMAKE_INSTALL_BINDIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/bin -DCMAKE_INSTALL_NAME_DIR=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7/lib -DCMAKE_POLICY_DEFAULT_CMP0025=NEW -DCMAKE_OSX_SYSROOT= -DCMAKE_FIND_FRAMEWORK=LAST -DCMAKE_STRIP=/nix/store/rwb88nwzbrsk2k0ldf1b5scd5yyrfgl9-cctools-binutils-darwin-11.1.0-973.0.1/bin/strip -DCMAKE_RANLIB=/nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/ranlib -DCMAKE_AR=/nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/ar -DCMAKE_C_COMPILER=clang -DCMAKE_CXX_COMPILER=clang++ -DCMAKE_INSTALL_PREFIX=/nix/store/nrsifj0zdqzv4pvv8mmssq9cphfjlf2z-libcxxabi-15.0.7 -DLLVM_ENABLE_RUNTIMES=libcxxabi -DLIBCXXABI_LIBCXX_INCLUDES=/nix/store/kd08z58zhhpvrww8a2f74dx1dr3frggb-libcxx-headers-15.0.7/include/c++/v1 -DCMAKE_CXX_COMPILER_WORKS=ON -DCMAKE_EXE_LINKER_FLAGS=-nostdlib -DCMAKE_SHARED_LINKER_FLAGS=-nostdlib
-- The C compiler identification is Clang 15.0.7
-- The CXX compiler identification is Clang 15.0.7
-- The ASM compiler identification is Clang with GNU-like command-line
-- Found assembler: /nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - failed
-- Check for working C compiler: /nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang
-- Check for working C compiler: /nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang - broken
CMake Error at /nix/store/inh6dp2sykgdnh6g4sblmng1s89vz08c-cmake-3.26.4/share/cmake-3.26/Modules/CMakeTestCCompiler.cmake:67 (message):
  The C compiler

    "/nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: /tmp/nix-build-libcxxabi-15.0.7.drv-0/libcxxabi-src-15.0.7/runtimes/build/CMakeFiles/CMakeScratch/TryCompile-NjGKoy

    Run Build Command(s):/nix/store/mqyp4gwyppmcykj3vk3ig9z1lgnvrrji-ninja-1.11.1/bin/ninja -v cmTC_f082c && [1/2] /nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang    -MD -MT CMakeFiles/cmTC_f082c.dir/testCCompiler.c.o -MF CMakeFiles/cmTC_f082c.dir/testCCompiler.c.o.d -o CMakeFiles/cmTC_f082c.dir/testCCompiler.c.o -c /tmp/nix-build-libcxxabi-15.0.7.drv-0/libcxxabi-src-15.0.7/runtimes/build/CMakeFiles/CMakeScratch/TryCompile-NjGKoy/testCCompiler.c
    [2/2] : && /nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang  -Wl,-search_paths_first -Wl,-headerpad_max_install_names -nostdlib CMakeFiles/cmTC_f082c.dir/testCCompiler.c.o -o cmTC_f082c   && :
    FAILED: cmTC_f082c
    : && /nix/store/dan3imi2bd0nv5vq10i53a3wqsfc5ya7-clang-wrapper-15.0.7/bin/clang  -Wl,-search_paths_first -Wl,-headerpad_max_install_names -nostdlib CMakeFiles/cmTC_f082c.dir/testCCompiler.c.o -o cmTC_f082c   && :
    ld: dynamic main executables must link with libSystem.dylib for architecture x86_64
    clang-15: error: linker command failed with exit code 1 (use -v to see invocation)
    ninja: build stopped: subcommand failed.





  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:3 (project)


-- Configuring incomplete, errors occurred!
error: builder for '/nix/store/ac8awd5c8fj8n0ya3bh5adkcz6mljijz-libcxxabi-15.0.7.drv' failed with exit code 1
error: 1 dependencies of derivation '/nix/store/3x3qxzgdbdzk706651wyhb3441sscr9p-compiler-rt-libc-15.0.7.drv' failed to build
error: 1 dependencies of derivation '/nix/store/n1ssw0pqiwb41gmi41zplqsfvl0zhcz0-clang-wrapper-15.0.7.drv' failed to build
building '/nix/store/ibn6vnib20jms9hwam74w88jb89awmlj-libtool.drv'...
error: 1 dependencies of derivation '/nix/store/59drfx8xm7c2lni83ipj4swiw6blwyaf-stdenv-darwin.drv' failed to build
error: 1 dependencies of derivation '/nix/store/q6acbkakd3ly1migpi614x514m7rmknk-swift-5.8.drv' failed to build
error: 1 dependencies of derivation '/nix/store/7dkwf3rkhlcjvm8pap8f12y99g86bjj4-mpv-0.36.0.drv' failed to build
error: 1 dependencies of derivation '/nix/store/0gdqb9idj7abnz5b3j6v9qpaj37l1cxy-mpv-with-scripts-0.36.0.drv' failed to build
```

--- a/pkgs/development/compilers/llvm/15/libcxxabi/default.nix
+++ b/pkgs/development/compilers/llvm/15/libcxxabi/default.nix
@@ -68,8 +68,7 @@ stdenv.mkDerivation rec {
   ] ++ lib.optionals (stdenv.hostPlatform.useLLVM or false) [
     "-DLLVM_ENABLE_LIBCXX=ON"
     "-DLIBCXXABI_USE_LLVM_UNWINDER=ON"
-  ] ++ lib.optionals ((stdenv.hostPlatform.useLLVM or false) ||
-                      (stdenv.hostPlatform.isDarwin && enableShared)) [
+  ] ++ lib.optionals ((stdenv.hostPlatform.useLLVM or false) || !stdenv.hostPlatform.isDarwin) [
     # libcxxabi's CMake looks as though it treats -nostdlib++ as implying -nostdlib,
     # but that does not appear to be the case for example when building
     # pkgsLLVM.libcxxabi (which uses clangNoCompilerRtWithLibc).

--- a/pkgs/development/compilers/llvm/16/libcxxabi/default.nix
+++ b/pkgs/development/compilers/llvm/16/libcxxabi/default.nix
@@ -68,8 +68,7 @@ stdenv.mkDerivation rec {
   ] ++ lib.optionals (stdenv.hostPlatform.useLLVM or false) [
     "-DLLVM_ENABLE_LIBCXX=ON"
     "-DLIBCXXABI_USE_LLVM_UNWINDER=ON"
-  ] ++ lib.optionals ((stdenv.hostPlatform.useLLVM or false) ||
-                      (stdenv.hostPlatform.isDarwin && enableShared)) [
+  ] ++ lib.optionals ((stdenv.hostPlatform.useLLVM or false) || !stdenv.hostPlatform.isDarwin) [
     # libcxxabi's CMake looks as though it treats -nostdlib++ as implying -nostdlib,
     # but that does not appear to be the case for example when building
     # pkgsLLVM.libcxxabi (which uses clangNoCompilerRtWithLibc).

--- a/pkgs/development/compilers/llvm/git/libcxxabi/default.nix
+++ b/pkgs/development/compilers/llvm/git/libcxxabi/default.nix
@@ -68,8 +68,7 @@ stdenv.mkDerivation rec {
   ] ++ lib.optionals (stdenv.hostPlatform.useLLVM or false) [
     "-DLLVM_ENABLE_LIBCXX=ON"
     "-DLIBCXXABI_USE_LLVM_UNWINDER=ON"
-  ] ++ lib.optionals ((stdenv.hostPlatform.useLLVM or false) ||
-                      (stdenv.hostPlatform.isDarwin && enableShared)) [
+  ] ++ lib.optionals ((stdenv.hostPlatform.useLLVM or false) || !stdenv.hostPlatform.isDarwin) [
     # libcxxabi's CMake looks as though it treats -nostdlib++ as implying -nostdlib,
     # but that does not appear to be the case for example when building
     # pkgsLLVM.libcxxabi (which uses clangNoCompilerRtWithLibc).

